### PR TITLE
This is the second commit 

### PR DESCRIPTION
**Pull request by leobeal**

### CODE REVIEW

**Comment by theravel** [comment]
Looks like you need to adjust git settings: commit is done from "another" author

**Comment by leobeal** [comment]
@theravel I guess I committed from my work laptop.  :) 

I would freshly create a new repo anyway for MM.

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
```
[the remark this answers is not in the document]
Comment (leobeal) [reply DependencyInjection/Configuration.php:1]: I had all code in `src`, but Symfony "Best Practices for Reusable Bundles" tells you to do so. 
At least this is why I understood. https://symfony.com/doc/current/bundles/best_practices.html#directory-structure

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      "MetroMarkets\\FFBundle\\Tests\\": "Tests"
+    }
+  },
+  "require": {
+    "symfony/config": "^5.0 || ^4.2",
+    "symfony/dependency-injection": "^5.0 || ^4.2",
+    "symfony/http-kernel": "^5.1.5 || ^4.2",
+    "symfony/yaml": "^5.0 || ^4.2",
+    "symfony/cache": "^5.0 || ^4.2",
+    "guzzlehttp/guzzle": "^7.0 || ^6.0"
```
[the remark this answers is not in the document]
Comment (leobeal) [reply composer.json:24]: sure it should!

File: Contract/ProviderInterface.php [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?php
```
[the remark this answers is not in the document]
Comment (leobeal) [reply Contract/ProviderInterface.php:1]: yep, NP

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+language: php
```
[the remark this answers is not in the document]
Comment (leobeal) [reply .travis.yml:1]: Travis file need to be on master.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+sudo: false
+cache:
+  directories:
+    - $HOME/.composer/cache/files
+    - $HOME/symfony-bridge/.phpunit
+env:
+  global:
+    - PHPUNIT_FLAGS="-v"
+    - SYMFONY_PHPUNIT_DIR="$HOME/symfony-bridge/.phpunit"
+matrix:
```
[the remark this answers is not in the document]
Comment (leobeal) [reply .travis.yml:11]: - about the coverage. I might have written somewhere, I was getting an error in phpunit, hard to understand. Disabled for now.
- php8 not yet available
- You are correct!
- Well for the same reason coverage will only be set for php 7.3 (or 7.4). We probably don't need to test all possible scenarios

File: FeatureFlagService.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        $this->provider = $provider;
+    }
+
+    public function isEnabled(string $key, $identifier = null): bool
+    {
+        return $this->provider->isEnabled($key, $identifier);
+    }
+
+    public function getProvider(): ProviderInterface
```
[the remark this answers is not in the document]
Comment (leobeal) [reply FeatureFlagService.php:22]: well, I though: Why not? But you're right! Removed it!

File: FeatureFlagService.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace MetroMarkets\FFBundle;
+
+use MetroMarkets\FFBundle\Contract\ProviderInterface;
+
+final class FeatureFlagService
```
[the remark this answers is not in the document]
Comment (leobeal) [reply FeatureFlagService.php:7]: `FeatureFlagService` is not final anymore ;)

File: Providers/RestProvider.php [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        $result = $this->fetchData($key);
+
+        $cacheItem->set($result);
+        $cacheItem->expiresAfter($this->cacheTTL);
+        $this->cache->save($cacheItem);
+
+        return $cacheItem->get();
+    }
+
+    private function fetchData(string $key)
```
[the remark this answers is not in the document]
Comment (leobeal) [reply Providers/RestProvider.php:57]: Correct!! But This is a first draft of the rest provider. Not a lot of thought was put here :)

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
+{
```
[the remark this answers is not in the document]
Comment (leobeal) [reply composer.json:1]: yes, it should :)

**Comment by theravel** [comment]
Leo, one more thing came to my mind:
If I use this bundle, and I want to write functional/integration test (which actually calls API endpoint which relies on feature flags) - there is currently no way to do so.

How I would see it:
- Such kind of testing should be possible without external dependencies. E.g. if I do contact-testing of my application, I want to run tests without external REST flags provider or ConfigCat.
- It could mean that I run such test with `APP_ENV=test`, then I could provide any kind of mocked flags (maybe YAML based?) in `services_test.yaml`

File: Providers/RestProvider.php (hunk 8 above)
[the remark this answers is not in the document]
Comment (theravel) [reply Providers/RestProvider.php:57]: What is your plan? (just to understand if I should push for it in this PR, or to expect another one :) )

File: FeatureFlagService.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (theravel) [reply FeatureFlagService.php:22]: I cannot see the change. Did you forget to push it?

**Pull request closed without merging**